docs/proposals/koordlet/20231208-netqos-terway.md

```diff
@@ -0,0 +1,371 @@
+The higher the node network bandwidth load, the lower the score, so as to ensure that the newly created pod can be scheduled
+to a node with relatively idle network bandwidth.
+```
+score = (node.capacity.netbandwidth - node.netbandwidth.used) * int64(framework.MaxNodeScore)) / node.capacity.netbandwidth
+```
+
+#### koord-descheduler
+ The `LowNodeLoad` rescheduler plugin needs to take into account the actual load on the node's network bandwidth when balancing. 
+ We need to add the `netBandwidth` threshold to the parameters of the `LowNodeLoad` plugin in koord-descheduler-config.yaml.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: descheduler-config
+  namespace: system
+data:
+  koord-descheduler-config: |
+    ...
+      - name: LowNodeLoad
+        args:
+          ...
+          lowThresholds:
+            netBandwidth: **
+          highThresholds:
+            netBandwidth: **
+```
+
+### usage:
+Cluster administrators can configure the [slo-controller-config.yaml](https://github.com/koordinator-sh/koordinator/blob/main/config/manager/slo-controller-config.yaml)
+to Configure cluster or node level network bandwidth, if the node bandwidth is not configured, the network bandwidth reported by the koordlet will be used,
+The default network bandwidth request percentage for each level is `l0:l1:l2=40%:30%:30%`, limit all 100% by default, administrators can configure it by themselves.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: slo-controller-config
+  namespace: system
+data:
+  colocation-config: |
+    {
+      "enable": true
+    }
+  resource-threshold-config: |
+    {
+      "clusterStrategy": {
+        "enable": true
+      }
+    }
+  resource-qos-config: |
+    {
+      "clusterStrategy": {
+        "lsrClass": {
+          "networkQOS": {
+            "enable": true,
+            "totalNetworkBandwidth": 1G,
+            "ingressRequest": 40,
+            "ingressLimit": 100,
+            "egressRequest": 40,
+            "egressLimit": 100,
+          },
+        },
+        "lsClass": {
+          "networkQOS": {
+            "enable": true,
+            "totalNetworkBandwidth": 1G,
+            "ingressRequest": 40,
+            "ingressLimit": 100,
+            "egressRequest": 40,
+            "egressLimit": 100,
+          },
+        },
+        "beClass": {
+          "networkQOS": {
+            "enable": true,
+            "totalNetworkBandwidth": 1G,
+            "ingressRequest": 30,
+            "ingressLimit": 100,
+            "egressRequest": 30,
+            "egressLimit": 100,
+          },
+        }
+      },
+      "nodeStrategies": [
+        {
+          "nodeSelector": {
+            "matchLabels": {
+              "kubernetes.io/kernel": "alios"
+            }
+          },
+          "beClass": {
+            "memoryQOS": {
+              "wmarkRatio": 90
+            }
+          }
+        }
+      ]
+    }
+```
+
```

Comment:
we need add a observability section to add some metrics of the running status of network qos plugin itself.
detailed implementation can be leave to future.